app/lib/geo.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Shared geographic calculations
+module Geo
+  class << self
+    ##
+    # Distance Between two location objects
+    #
+    # Accepts two objects of any type that contain a
+    # lat and long method.
+    #   location        object
+    #   other_location  object
+    #
+    # Returns the distance between these two objects
+    def distance_between(location, other_location)
+      return '' if location.nil? || other_location.nil?
+
+      haversine_distance_between(location.lat,
+                                 location.long,
+                                 other_location.lat,
+                                 other_location.long, true)
+    end
+
+    ##
+    # Haversine Distance Calculation
+    #
+    # Accepts the latitude and longitude of two locations
+    #   lat1  float
+    #   lon1  float
+    #   lat2  float
+    #   lon2  float
+    #
+    # Returns the distance between these two
+    # points in either miles or kilometers
+    def haversine_distance_between(lat1, lon1, lat2, lon2, miles)
+      # Calculate radial arcs for latitude and longitude
+      d_lat = (lat2 - lat1) * Math::PI / 180
+      d_lon = (lon2 - lon1) * Math::PI / 180
+
+      haversine_a = haversine_a(d_lat, d_lon, lat1, lat2)
+
+      c = haversine_c(haversine_a)
+
+      (6371 * c * (miles ? 1 / 1.6 : 1)).round(2)
+    end
+
+    def haversine_a(d_lat, d_lon, lat1, lat2)
+      Math.sin(d_lat / 2)**2 +
+        Math.cos(lat1 * Math::PI / 180) *
+        Math.sin(d_lon / 2)**2 *
+        Math.cos(lat2 * Math::PI / 180)
+    end
+
+    def haversine_c(haversine_a)
+      2 * Math.atan2(Math.sqrt(haversine_a), Math.sqrt(1 - haversine_a))
+    end
+
+    def validate_coordinate_values(lat, long)
+      return false unless (lat >= -90 && lat <= 90) &&
```

Comment:
I added that validation as a guard for Geo.distance_between().  Perhaps this could be something to discuss in the stand-up with David as to whether it is an issue.

Reply:
> As long as we're validating values do you think it makes sense to limit these to United States lat/longs? I could go either way on this, up to you.

I would say there is value in this. It can simplify screening on the front end. Current potential market for FreshTrak is Continental US + Alaska + Hawaii + Puerto Rico. Based on Phil's comment, appears to already be addressed.

Reply:
It is not addressed.  Based on a later slack comment from David, I think we have agreement this will not be included in this PR but will be prioritized as a separate issue.